### ISSUE TITLE
OpenSSL 3.x compatiblity

### ISSUE THREAD

**Issue by michaelklishin** [bloomberg/amqpprox#95]
When I try to build `amqpprox`, I run into a conan failure that says that its `openssl@1.1k` package cannot be provisioned.

I tried both macOS and Fedora Linux, and it turns out, on both machines I run OpenSSL/LibreSSL 3.x.

Am I correct to assume that `amqpprox` currently requires OpenSSL 1.1?

**Comment by michaelklishin** [comment]
I'd be happy to move this to a discussion if they were enabled for this repo ;)

**Comment by adamncasey** [comment]
Thanks for reporting! It should build with OpenSSL 3, I'll figure out whether it's just a packaging issue or something more involved.

**Comment by adamncasey** [comment]
I agree

```
openssl/1.1.1k: Downloaded recipe revision d244a9c61b38140fc03ffddcfdfc3547
Graph root
    conanfile.txt: /home/adam/amqpprox/buildfiles/conan/conanfile.txt
Requirements
    boost/1.76.0#84b03970d0b0284bb317f75446dc7056 - Downloaded (conancenter)
    bzip2/1.0.8#411fc05e80d47a89045edc1ee6f23c1d - Downloaded (conancenter)
    libbacktrace/cci.20210118#ec1aa63bbc10145c6a299e68e711670c - Downloaded (conancenter)
    zlib/1.2.13#13c96f538b52e1600c40b88994de240f - Downloaded (conancenter)
Build requirements
    b2/4.9.3#5445210661950ce568cfd76158d9e117 - Downloaded (conancenter)
*********************************************************
Recipe 'openssl/1.1.1k' seems broken.
It is possible that this recipe is not Conan 2.0 ready
If the recipe comes from ConanCenter check: https://conan.io/cci-v2.html
If it is your recipe, check if it is updated to 2.0
*********************************************************

ERROR: Package 'openssl/1.1.1k' not resolved: openssl/1.1.1k: Cannot load recipe.
Error loading conanfile at '/home/adam/.conan2/p/opensf766fd3864f72/e/conanfile.py': Unable to load conanfile in /home/adam/.conan2/p/opensf766fd3864f72/e/conanfile.py
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/adam/.conan2/p/opensf766fd3864f72/e/conanfile.py", line 4, in <module>
    from conans import ConanFile, AutoToolsBuildEnvironment, tools
ImportError: cannot import name 'ConanFile' from 'conans' (/home/adam/.local/lib/python3.10/site-packages/conans/__init__.py)
```

Suspect we can change the metadata to require at least version 1.1.1k

**Comment by adamncasey** [comment]
It looks like this has broken as part of the recent conan 2.0 release. python -mpip install "conan<2.0" seems to make more progress.

Either way I think we should support building on at least openssl 1.1 though, rather than pinning to a particular version

**Comment by adamncasey** [comment]
Since #97 I've been able to build amqpprox natively on a mac without problems. #98 will fix the openssl 3.0+ build issue, and importantly stop pinning to an out of date openssl version
